Задача: Разбираем склад Amazon.

piligrim-curious opened the pull request:
**Product**: Класс продуктов;
**ProductManager**: Менеджер продуктов.

piligrim-curious committed:
Задача: Разбираем склад Amazon.
Product: Класс продуктов;
ProductManager: Менеджер продуктов.

piligrim-curious committed:
Задача: Разбираем склад Amazon.
Product: Класс продуктов;
ProductManager: Менеджер продуктов.

SnurnicynKonstantin commented on src/main/java/school/faang/amazon_warehouse/Product.java:
```diff
@@ -0,0 +1,20 @@
+    private final String name;
+    private final Category category;
+
+    public Product(Integer id, String name, Category category) {
```
Можно использовать аннотацию @AllArgsConstructor что бы не создавать самому конструктор. Он будет сгенерирован автоматически

SnurnicynKonstantin commented on src/main/java/school/faang/amazon_warehouse/Product.java:
```diff
@@ -0,0 +1,20 @@
+@ToString
+@Getter
+public class Product {
+    public enum Category { FOOD, ELECTRONICS, CLOTHING, OTHER }
```
enum лучше в отдельный файл выносить

SnurnicynKonstantin commented on src/main/java/school/faang/amazon_warehouse/ProductManager.java:
```diff
@@ -0,0 +1,107 @@
+            }
+        }
+
+        return new ArrayList<>(productListByCategory);
```
new ArrayList<>( не обязательно здесь. 
Метод должен возвращать List<Product>, а productListByCategory как раз таким типом и является. 
Так что можно просто `return productListByCategory;`

piligrim-curious committed:
Задача: Разбираем склад Amazon.
Product: Класс продуктов;
ProductManager: Менеджер продуктов;
[+] Category: Категории товаров.

Изменения:
(Category.java)
* Добавлен класс - Category.java;
(Product.java)
* Добавлена нотация Lombok для автогенерации конструктора класса;
* Удалён enum Category;
(ProductManager.java)
* Упрощён возврат значения функции findProductsByCategory().

SnurnicynKonstantin reviewed (approved):
Здорово, что освоил enum и новую аннотацию
